helpers/TaskQueue.js

```diff
@@ -46,40 +66,42 @@ class TaskQueue {
    * @param {object} time - DateTime object of timestamp at which to resume
    */
   async pauseUntil(time) {
-    // Lock mutex while altering this.isPaused
-    const unlock = await this.mutex.lock();
-
     // If queue is already paused, do nothing
     if (this.isPaused) {
       return;
     }
 
-    clearInterval(this.intervalId);
+    // Pause execution
     this.isPaused = true;
-
-    unlock();
-
     if (time > Date.now()) {
       await new Promise((r) => { setTimeout(r, time - Date.now()); });
     }
-    // Begin new execution interval.
-    this._start();
+
+    // Resume execution.
+    this.isPaused = false;
+
+    // Skip potential unnecessary delay
+    this._skipIntervalDelayIfApplicable();
```

Comment:
Are we sure this doesn't need to occur in a locked scope?

Reply:
skipIntervalDelay locks itself, so we don't want to lock around it. I don't think it's a problem to have the call be unlocked.